### PR TITLE
OTP entry form instead of window alert

### PR DESCRIPTION
**Pull request by whaught**
Fixes https://github.com/google/exposure-notifications-verification-server/issues/545

## Proposed Changes

* Create a custom html pin form for login and registerphone
* Resend SMS link is disabled and enables on a 15s timer.
* Use the close-icon to return to previous
* Some css tweaks

pinform

register

**Release Note**

```release-note
Use custom html for entering SMS pin
```

### CODE REVIEW

File: cmd/server/assets/header.html [hunk 1]
```diff
@@ -34,6 +34,11 @@
     cursor: pointer;
     text-decoration: none;
   }
+
+  a.disabled {
+  color: lightgray;
```
Comment (sethvargo) [inline cmd/server/assets/header.html:39]: Nit indent

File: cmd/server/assets/header.html [hunk 2]
```diff
@@ -229,25 +234,27 @@ <h6 class="dropdown-header">Actions</h6>
...
-  {{$msg}}
-</div>
-{{end}}
-{{range $msg := .flash.Alerts}}
-<div class="alert alert-success alert-dismissible fade show" role="alert">
-  {{$msg}}
-  <button type="button" class="close" data-dismiss="alert" aria-label="Close">
-    <span aria-hidden="true">&times;</span>
-  </button>
+<div id="alerts">
```
Comment (sethvargo) [inline cmd/server/assets/header.html:237]: I can see us accidentially conflicting with this in the future. Can we make this `alerts-container` or something more specific of an ID?

File: cmd/server/assets/login/_loginscripts.html [hunk 3]
```diff
@@ -38,3 +38,28 @@
   }
 </script>
 {{end}}
+
+{{define "pindiv"}}
```
Comment (sethvargo) [inline cmd/server/assets/login/_loginscripts.html:42]: Maybe "login/pindiv" so it's namespaced?

File: cmd/server/assets/login/_loginscripts.html [hunk 4]
```diff
@@ -38,3 +38,28 @@
...
+  <div class="card-header">
+    SMS Pin
+    <button type="button" class="close" aria-label="Close" id="pinClose">
+      <span aria-hidden="true">&times;</span>
+    </button>
+  </div>
+  <div class="card-body">
+    <form id="pinForm" class="floating-form" action="/" method="POST">
+      <div class="form-label-group">
+        <input type="text" id="pin" name="pin" class="form-control" placeholder="Pin" autocomplete="one-time-code" required autofocus />
```
Comment (sethvargo) [inline cmd/server/assets/login/_loginscripts.html:53]: We're using PIN and OTP interchangeably here - should standardize on one if possible.

Comment (sethvargo) [reply cmd/server/assets/login/_loginscripts.html:53]: And, if we go with PIN, we should always capitalize it as "PIN"

File: cmd/server/assets/login/login.html [hunk 5]
```diff
@@ -15,7 +15,8 @@
     <div class="d-flex vh-100">
       <div class="d-flex w-100 justify-content-center align-self-center">
         <div class="col-sm-6">
-          <div class="card shadow-sm">
+
+          <div class="card shadow-sm" id="loginDiv">
```
Comment (sethvargo) [inline cmd/server/assets/login/login.html:19]: CSS ids should be kebab-cased

File: cmd/server/assets/login/login.html [hunk 6]
```diff
@@ -114,6 +120,49 @@
...
+          });
+      });
+
+      $pinClose.on('click', function(event) {
+        $submit.prop('disabled', false);
+        $loginDiv.show();
+        $pinDiv.hide();
+      });
+
+      $resendPin.on('click', function(event) {
```
Comment (sethvargo) [inline cmd/server/assets/login/login.html:148]: Do you need to prevent default here?

Comment (whaught) [reply cmd/server/assets/login/login.html:148]: it's a link with no href, not a submit button?

**Pull request merged**